pkg/didcomm/common/service/messenger.go

```diff
@@ -28,12 +28,25 @@ type Messenger interface {
 	// ReplyTo replies to the message by given msgID.
 	// Keeps threadID in the *decorator.Thread.
 	// Using this function means that communication will be on the same thread.
-	ReplyTo(msgID string, msg DIDCommMsgMap)
+	ReplyTo(msgID string, msg DIDCommMsgMap) error
 
 	// Send sends the message by starting a new thread.
-	Send(msg DIDCommMsgMap, myDID, theirDID string)
+	Send(msg DIDCommMsgMap, myDID, theirDID string) error
```

Comment:
I'm confused on why the msg argument in these methods is not of the interface type DIDCommMsg.

Reply:
@troyronda , may be because, DIDCommMsg doesn't exactly looks like interface for DID Communication message type, 
eg: presence of `Decode(v interface{}) error` in  DIDCommMsg interface forces sender to implement Decode function

Reply:
The interface should be provided only when someone wants to do `read` operations. There is no need to implement another did message interface and should not be. Messages across the framework should be `type DIDCommMsgMap map[string]interface{}` which covers any type of possible message.

Reply:
So is the intent to delete `service.DIDCommMsg` and have consumers define their dependency?

Reply:
no, my assumption is that everything should be a map interface. But it can be easily changed. if you really need to define a custom implementation, then service.DIDCommMsg should provide one more method e.g ToMapInterface() and it will be used at least for the messenger.

Reply:
but my intention was to create a one generic message type

Reply:
@soluchok can you provide an example of how this msg map will be modified? And by whom? I think I understand that the `Messenger` will be modifying it for low-level didcomm stuff, but I'm curious to know if the higher-level protocols will also be modifying it? Can a protocol add a hypothetical `~payment` decorator?